src/Listener/TerminateListener.php

```diff
@@ -165,4 +179,58 @@ private function validateInputMap(array $inputMap, string $commandClass): void
             Assert::allString($value, 'Malformed input mapper for ' . $commandClass);
         }
     }
+
+    /**
+     * @psalm-return non-empty-string
+     */
+    private function getVendorDirectory(): string
+    {
+        $basePath     = realpath(getcwd());
+        $composerJson = file_get_contents($basePath . '/composer.json');
+        Assert::string($composerJson);
+
+        $composer = json_decode($composerJson, true);
+        Assert::isMap($composer);
+
+        $vendorDir = $composer['config']['vendor-dir'] ?? $basePath . '/vendor';
+        Assert::string($vendorDir);
+        Assert::directory($vendorDir);
+
+        $vendorDir = $this->normalizePath(realpath($vendorDir));
+        return rtrim($vendorDir, '/') . '/';
+    }
+
+    /**
+     * @psalm-param class-string $class
+     */
+    private function matchesLocalNamespace(string $class, string $vendorDir): bool
```

Comment:
I would suggest to change the method name, as we are not matching the vendor dir not the namespace